docs/05-dev-tools/oracles/umbrellanetwork.md

```diff
@@ -10,19 +10,778 @@ Umbrella Network provides decentralized and scalable oracles designed for high-p
 
 The platform combines a decentralized architecture with seamless integration options to deliver reliable data feeds. It is compatible with 12 major blockchains, including Rootstock, Ethereum, BNB Chain, ensuring extensive support for various blockchain projects.
 
-Umbrella Network offers data feeds for thousands of cryptocurrencies, crypto options, stocks, and real-world assets. Below is an overview of its interactive API documentation structure:
-
-
-## **Key Features**
-1. **Price Feeds**:
-   - Real-time, decentralized price data for crypto assets and other markets.
-   - Custom feeds available upon request.
-2. **Rollup Data**:
-   - High-performance data handling through the Rollup Data SDK, offering Merkle proof-backed data validation.
-3. **On-Chain Data**:
-   - Consensus-based updates for price deviations and other data triggers.
-4. **Random Number Generation**:
-   - A secure RNG solution for blockchain applications.
+## Requirements
+[foundry](https://dev.rootstock.io/dev-tools/dev-environments/foundry/)
+
+
+## Umbrella Network Architecture
+Umbrella Network is made up of validator nodes that fetch off-chain data and prove that the data is valid and unchanged.<br></br>
+Nodes stake a certain amount of UMB token to act as collateral in case of a malicious actor.<br></br>
+Many independent validator nodes pull high quality data from defined APIs and submit it offchain to Umbrella's layer-2 network.<br></br>
+Several validator nodes are then elected as leaders, which means they are in charge of agreeing on the final value (consensus) of the data to be submitted to the various supported blockchains.<br></br>
+Once consensus has been reached, a cryptographic proof is added to the data to prevent tampering.<br></br>
+The data is then broadcast to supported networks and updated in the appropriate smart contracts.<br></br>
+Usually, data from high quality sources has minimal to no deviation. A big difference in one oracle's value from the rest of the values indicates a faulty or malicious validator which then gets their stake slashed.<br></br>
+
+
+## Read data from Umbrella Network smart contracts using Foundry
+There are several smart contracts involved when reading data from Umbrella Network, which we will briefly describe and use:
+- Umbrella Registry
+- UmbrellaFeeds
+- Reader contracts
+
+To read data directly, only the first two are used.<br></br>
+In some cases, however, you might want to use another contract as a proxy to read the same data, for which you need to deploy another contract to do that (reader contract).
+
+
+The actual data is found in the `Umbrella Feeds` smart contract. However, you must first get the address from the `Umbrella Registry` contract.
+
+:::info[Info]
+It is recommended to run the commands below on mainnet because
+- We are only reading from the contracts, no gas will be used unless you are deploying your own [reader contract](#reader-contracts).
+- Most price feeds are not available on testnet.
```

Comment:
List the price feeds available on mainnet for Rootstock: https://umbrella-network.github.io/technical-documentation/umbrella-network/docs/price-feeds.html#mainnets